python/lsst/daf/butler/registry/databases/postgresql.py

```diff
@@ -113,8 +115,9 @@ def transaction(self, *, interrupting: bool = False, savepoint: bool = False,
 
     def _lockTables(self, tables: Iterable[sqlalchemy.schema.Table] = ()) -> None:
         # Docstring inherited.
-        for table in tables:
-            self._connection.execute(f"LOCK TABLE {table.key} IN EXCLUSIVE MODE")
+        with self._connection() as connection:
```

Comment:
So this context manager frees the connection object but doesn't have any effect on the table locking?

Reply:
The table locking will automatically end when the current transaction ends, and closing the connection will (I think) do that.

Reply:
I believe this relies on the fact that `_lockTables()` is called from `transaction()` and in that case connection only closes when that outermost transaction closes. I should have probably used `self._session_connection` here instead of `self._connection()` even though they produce identical result. Whole connection transaction management is a bit messy, sqlalchemy 2.0 tries to simplify it but it does not map too well into what we are trying to do. Maybe we need to review it at some point. And in this particular case I think I can improve it just by passing connection as an argument to this method.

Reply:
I updated `_lockTables()`  argument list to pass `connection` directly from transaction(), this should avoid possible issues with connection lifetime.